partcad/src/partcad/part_factory_enrich.py

```
@@ -89,13 +101,43 @@ def __init__(self, ctx, source_project, target_project, config):
-            )
+            if "with" in part.config:
+                if "parameters" not in augmented_config:
+                    raise Exception(
+                        "Attempting to parametrize a part that has no parameters: %s" % str(augmented_config)
+                    )
+                for param in part.config["with"]:
+                    if param not in augmented_config["parameters"]:
+                        raise Exception(
+                            "Attempting to parametrize a part with an unknown parameter: %s:%s: %s"
+                            % (self.source_project_name, self.source_part_name, param)
+                        )
```

Ruff (0.8.2)

Create your own exception

(TRY002)